Automattic/jetpack
# Simple Payments: prevent XSS on title and content with the Gutenberg Block

With the addition of the Gutenberg Simple Payments block, we are adding yet another way to create Simple Payment Products by using the `wp v2` API endpoint to create posts using the `jp_pay_product` custom post type.

In WordPress.com case, there are no problems attempting to send html code on the title or description of a product because the `kses` filters are enabled for all users. But in the case of Jetpack, if the user has the `unfiltered_html` capability, the filters to sanitize title and content are not added, opening the window to XSS attacks when using the Gutenberg block.

This PR applies ~~`wp_kses`~~`esc_html` to the `title` and `content` output when processing the Simple Payments shortcode to prevent XSS attacks on those fields using the Gutenberg Block.

![screen shot 2018-11-16 at 12 25 05](https://user-images.githubusercontent.com/233601/48630335-cb4f3000-e99a-11e8-93a4-03a069bf14e1.png)

| Before | After |
| - | - |
| ![before](https://user-images.githubusercontent.com/233601/48630614-719b3580-e99b-11e8-979e-36a450fb2591.gif)|![after](https://user-images.githubusercontent.com/233601/48630629-7a8c0700-e99b-11e8-924b-ac1caeccb178.gif)|

#### Testing instructions:





* [Setup a Jurassic Ninja Test Site](https://jurassic.ninja/create/?shortlived&gutenpack&gutenberg&jetpack-beta&branch=fix/simple-payments-title-content-kses).
* Connect Jetpack with a Premium or Professional Plan.
* Create a new Post.
* Add the Simple Payments Block.
* Create a Product attempting to insert malicious code (for example `<script>alert('foo');</script>`) on the title or description of the product.
* Publish the post.
* View the published post.

Without the changes on this PR, a JavaScript alert should show, and the title (or description) of the product should be missing. When applying these changes, the offending HTML tags are removed and no alert should appear.
